File: python/Communication/ws_server.py
import asyncio
import websockets
import signal

async def server(websocket, path):
    print("A new client connected")
    try:
        while True:
            message = await websocket.recv()
            print(f"Received: {message}")
            await websocket.send("Hello, client!")
    except websockets.exceptions.ConnectionClosed:
        print("Client disconnected")

start_server = websockets.serve(server, "localhost", 8080)

# 是一个关不掉的窗口
asyncio.get_event_loop().run_until_complete(start_server)
asyncio.get_event_loop().run_forever()

File: javascript/ws_server.js
const WebSocket = require('ws');

const wss = new WebSocket.Server({ port: 8080 });

const jsonData = {
  "dmms_m": [
      {
          "device_id_m": 1,
          "t_year_m": 2024,
          "t_month_m": 3,
          // 其他字段请根据实际情况填写
      }
  ]
};

wss.on('connection', function connection(ws) {
  console.log('A new client connected');

  ws.on('message', function incoming(message) {
    console.log('Received: %s', message);
  });

  // ws.send('Hello, client!');
  ws.send(JSON.stringify(jsonData));
});
